adagrams/game.py

```diff
@@ -1,13 +1,148 @@
+        for single_letter in range(letter_distribution):
+            available_letters.append({'letter':letter, 'is_not_available':False})
+    
+    return available_letters
+     
 def uses_available_letters(word, letter_bank):
-    pass
+    letters_in_word = list(word.upper())
+    is_valid = False
+
+    for letter in word:
+        same_letter_count = letters_in_word.count(letter) <= letter_bank.count(letter)
+        
+        if word.count(letter) > 0 and same_letter_count:
+            continue
+        else:
+            return is_valid
+    
+    is_valid = True
+
+    return is_valid
 
 def score_word(word):
-    pass
+    SCORE_CHART = {
+        1 : ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+        2 : ['D', 'G'],
+        3 : ['B', 'C', 'M', 'P'],
+        4 : ['F', 'H', 'V', 'W', 'Y' ],
+        5 : ['K'],
+        8 : ['J', 'X' ],
+        10 : ['Q', 'Z']
+    }
+    BONUS_POINTS = 8
+    score = 0
+    word_case_insensitive = word.upper()
+
+    for letter in word_case_insensitive:
+        for points, letters in SCORE_CHART.items():
+            if letter in letters:
+                score += points
+
+    if len(word) >= 7:
+        score += BONUS_POINTS
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    words_and_scores = build_words_and_scores_dict(word_list)
+    top_score = 0
+    highest_scores = []
+    winning_word_and_score = []
+
+    for current_word, current_score in words_and_scores.items():
+        if top_score == 0 or current_score == top_score:
+            top_score = current_score
+            highest_scores.append([current_word, current_score])
+        elif current_score > top_score:
+            top_score = current_score
+            highest_scores[0] = [current_word, current_score]
+    
+    if len(highest_scores) > 1:
+        winning_word_and_score = check_for_a_tie(highest_scores)
+    else:
+        winning_word_and_score = highest_scores[0]
+
+    return tuple(winning_word_and_score)
+
+def get_index_of_letter():
+    return randint(0, 97)
+
+def reset_available_letters(available_letters):
+    for letter in range(len(available_letters)):
+        available_letters[letter]['is_not_available'] = False
+
+def build_words_and_scores_dict(word_list):
+    words_with_scores_dict = {}
+
+    for current_word in word_list:
+        current_score = score_word(current_word)
+        words_with_scores_dict[current_word] = current_score
+    
+    return words_with_scores_dict
+
+def check_for_a_tie(words_and_scores):
+    winning_word_and_score = []
+    words_with_same_len = [] 
+    
+    for current_word_and_score in range(len(words_and_scores)):
+        if not winning_word_and_score:
+            winning_word_and_score = words_and_scores[current_word_and_score]
+        if len(words_and_scores[current_word_and_score][0]) == 10:
+            winning_word_and_score = words_and_scores[current_word_and_score]
+            break
```

Comment:
What if instead of using `break`, we just returned?
```suggestion
            return words_and_scores[current_word_and_score]
```